waku/v2/protocol/waku_relay.nim

```diff
@@ -12,10 +13,26 @@ import
 logScope:
     topics = "wakurelay"
 
-const WakuRelayCodec* = "/vac/waku/relay/2.0.0-beta2"
+const
+  WakuRelayCodec* = "/vac/waku/relay/2.0.0-beta2"
+  DefaultKeepAlive = 5.minutes # 50% of the default chronosstream timeout duration
 
 type
   WakuRelay* = ref object of GossipSub
+    keepAlive*: bool
+
+proc keepAlive*(w: WakuRelay) {.async.} =
+  while w.keepAlive:
+    # Keep all mesh peers alive when idle
+    trace "Running keepalive"
+
+    for topic in w.topics.keys:
```

Comment:
Curious to know how the idle topics are identified in here?

Reply:
For now we keep all topics alive, whether idle or not. This is not ideal, so I might add some per-topic activity indicator. It could be as simple as an `activity` boolean that is set to `true` on any activity and resets to `false` whenever the `keepAlive` runs. The `keepAlive` will then only ping on topics with no activity since the last time the process kicked off.